packages/plugin-core/src/telemetry.ts

```diff
@@ -1,87 +1,24 @@
-import { ConfigUtils, DWorkspaceV2, VSCodeEvents } from "@dendronhq/common-all";
-import { SegmentClient, TelemetryStatus } from "@dendronhq/common-server";
-import _ from "lodash";
-import * as vscode from "vscode";
-import { ExtensionProvider } from "./ExtensionProvider";
+import { ConfigUtils, DWorkspaceV2 } from "@dendronhq/common-all";
+import { SegmentClient } from "@dendronhq/common-server";
 import { Logger } from "./logger";
-import { AnalyticsUtils } from "./utils/analytics";
-
-export function isVSCodeTelemetryEnabled(): boolean | undefined {
-  // `isTelemetryEnabled` only seems to be available on VSCode 1.55 and above.
-  // Our dependencies are currently set up for 1.54, so typescript does not understand that it exists.
-  try {
-    return (vscode.env as any).isTelemetryEnabled;
-  } catch (err) {
-    Logger.error({ msg: "Error when trying to access telemetry preference" });
-    return undefined;
-  }
-}
 
 /** Creates a SegmentClient for telemetry, if enabled, and listens for vscode telemetry settings to disable it when requested. */
 export function setupSegmentClient(ws: DWorkspaceV2, cachePath?: string) {
   function instantiateSegmentClient() {
```

Comment:
The resulting code is a little funky: we have a function that basically wraps a single function (both named essentially the same thing).  Can we just delete the inner function and move it into the try block?